### DISCUSSION TITLE
How do you write & operate on unicode character literals in C?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#3179] [Q&A] [answered]
I recently embarked on rewriting my scanner from C++ to C (see PR: https://github.com/tlaplus-community/tree-sitter-tlaplus/pull/99). This was mostly a smooth process but then I hit an issue that I recall actually originally led me to use C++ instead: MSVC on Windows treats Unicode character literals differently than GCC or LLVM do. You can see this in this CI run, where the tests only fail on Windows and only the Unicode tests that touch the external scanner fail: https://github.com/tlaplus-community/tree-sitter-tlaplus/actions/runs/8257874149/job/22589163178?pr=99

So how does one write C code that handles Unicode character literals in a cross-platform way? For example, here is a line of code that works how you'd expect on macOS/Linux (LLVM and GCC) but not Windows & MSVC:
```C
int32_t lookahead = lexer->lookahead;
if (L'∧' == lookahead) ...
```

**Comment by ahelwer** [comment]
Answering my own question here, you have to use the `L'\u2227'` form for characters to make sure MSVC accepts it. It's annoying but whatever, it works.

**Comment by amaanq** [comment, marked as answer]
i think the `/utf-8` flag on windows might fix that as well, and does seem to fix #3222 too, typical windows
